model/context.go

```diff
@@ -0,0 +1,79 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package model
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/elastic/apm-server/utility"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+func HttpFields(http common.MapStr) common.MapStr {
+	var source = func(dottedKeys string) interface{} {
+		return utility.Get(http, dottedKeys)
+	}
+
+	destination := common.MapStr{}
+	utility.Add(destination, "version", source("request.http_version"))
+	utility.DeepAdd(destination, "request.method", source("request.method"))
+
+	utility.DeepAdd(destination, "request.method", source("request.method"))
+	utility.DeepAdd(destination, "request.body.content", source("request.body"))
+	utility.DeepAdd(destination, "request.headers.cookies.parsed", source("request.cookies"))
+	utility.DeepAdd(destination, "request.headers.cookies.original", source("request.headers.cookie"))
+	utility.DeepAdd(destination, "request.headers.user-agent", source("request.headers.user-agent"))
+	utility.DeepAdd(destination, "request.headers.content-type", source("request.headers.content-type"))
+	utility.DeepAdd(destination, "request.env", source("request.env"))
+	utility.DeepAdd(destination, "request.socket", source("request.socket"))
+	utility.DeepAdd(destination, "response.finished", source("response.finished"))
+	utility.DeepAdd(destination, "response.status_code", source("request.status_code"))
+	utility.DeepAdd(destination, "response.headers.content-type", source("response.headers.content-type"))
+	utility.DeepAdd(destination, "response.headers.sent", source("request.headers_sent"))
+
+	return destination
+}
+
+func UrlFields(http common.MapStr) common.MapStr {
```

Comment:
I don't think this should be called `http` as most of the information is just fetched from `context`, I'd rather call this `input` or `fields`.